tests/framework/utils.py

```diff
@@ -89,6 +89,22 @@ def get_files_from(find_path: str, pattern: str, exclude_names: list = None,
     return found
 
 
+def get_free_mem_ssh(ssh_connection):
+    """
+    Get how much free memory a guest sees, over an ssh connection.
+
+    :param ssh_connection: connection to the guest
+    :return: available mem column output of 'free'
+    """
+    _, stdout, stderr = ssh_connection.execute_command('free')
```

Comment:
Wouldn't `cat /proc/meminfo | grep MemAvailable` be a more reliable method of extracting the data? 
`free` uses `meminfo` behind the scenes, so probably we could avoid the for loop:

def get_free_mem_ssh(ssh_connection):
    """
    Get how much free memory a guest sees, over an ssh connection.
    :param ssh_connection: connection to the guest
    :return: available mem column output of 'free'
    """
    _, stdout, stderr = ssh_connection.execute_command('cat /proc/meminfo | grep MemAvailable')
    assert stderr.read() == ''
    
    # Split "MemAvailable:   123456 kB" and validate it
    meminfo_data = stdout.split(" ")
    if len(meminfo_data) == 3:
            return free_kb[1]
        
    raise Exception('Available memory not found in `free` output')

Reply:
I think the `free` output is also pretty standard, but I have no strong opinion on this. I could change this if it makes a difference.

Reply:
I vote for `/proc/meminfo` too, particularly if in the future we might care about finer grained stats and since it's not a huge change (hopefully :smile:)

Reply:
Used `/proc/meminfo`.